irohad/multi_sig_transactions/impl/mst_processor.cpp

```diff
@@ -16,3 +16,28 @@
  */
 
 #include "multi_sig_transactions/mst_processor.hpp"
+
+namespace iroha {
+
+  MstProcessor::MstProcessor() { log_ = logger::log("MstProcessor"); }
+
+  void MstProcessor::propagateTransaction(ConstRefTransaction transaction) {
+    std::lock_guard<std::mutex> lock{this->mutex_};
```

Comment:
In my opinion, we should add some explanation about why this lock is needed, what shared data is, which threads can have simultaneous access to them and so on. Such description will make reading and especially debugging this part of code a way more easier.